lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Transactions.hs

```diff
@@ -130,9 +130,15 @@ spec = do
     it "TRANS_EXTERNAL_CREATE_03 - proper single output transaction and \
        \wrong binary format" $ \ctx -> do
         let toSend = 1 :: Natural
-        (ExternalTxFixture _ _ _ txWits) <- fixtureExternalTx ctx toSend
+        (ExternalTxFixture _ _ _ tx) <- fixtureExternalTx ctx toSend
+
+        let run = SignedTxBinary . BL.toStrict . runPut
+        let putBin (SignedTxBinary bs) = putByteString bs
+        let addHeader = run . (withHeader FragmentTransaction) . putBin
+
+        let wrongTx = addHeader tx
         let baseOk = Base16
-        let arg = T.unpack $ encodeTx txWits FragmentInitial baseOk
+        let arg = T.unpack $ txToBase wrongTx baseOk
```

Comment:
Something is wrong in this update. The test itself seems to have changed?

Reply:
Previously the fixture provided an intermediate `(Tx, [TxWitness])` which we could easily encode incorrectly (with the `FragmentInitial` header instead of `FragmentTransaction`).

Now the fixture already provides a correct binary. We cannot easily _change_ the fragment header to an `FragmentInitial`, so I opted to wrap the fragment an extra time with `FragmentTransaction`.

Reply:
Hmm okay. I think the whole thing is still somewhat overkill. This whole test could really be: 

```hs
$ \ctx -> do
    let invalidArg = "0000"
    (Exit code, Stdout out, Stderr err) <- postExternalTransactionViaCLI @t ctx [invalidArg]
    err `shouldContain` errMsg400MalformedTxPayload
    out `shouldBe` mempty
    code `shouldBe` ExitFailure 1
```

What do you think?